lib/baremetalbasetest.pm

```diff
@@ -0,0 +1,269 @@
+
+    $self->SUPER::post_run_hook();
+}
+
+=head2 post_fail_hook
+
+ post_fail_hook();
+
+When the test module fails, this method will be called.
+It will try to fetch some logs from the SUT.
+Test modules (or their intermediate base classes) may overwrite
+this method to export certain specific logfiles and call the
+base method using C<$self-E<gt>SUPER::post_fail_hook;> at the end.
+
+=cut
+sub post_fail_hook {
+    my ($self) = @_;
+
+    # usually, we are on a serial terminal, but we could of course
+    # also run on a VNC session, so support this by running the
+    # post fail hook from the base class
+    $self->SUPER::post_fail_hook();
+
+    # unlock the host if it was locked. Make sure not to keep the
+    # host locked for longer that needed.
+    if (host_is_locked()) {
+        host_unlock();
+    }
+
+}
+
+=head2 poweron
+
+ poweron()
+
+Power on the machine if the backend in use is supported. 
+=cut
+sub poweron {
+    if (check_var('BACKEND', 'ipmi')) {
+        ipmitool("chassis power on");
+        while (1) {
+            sleep(3);
+            my $stdout = ipmitool('chassis power status');
+            last if $stdout =~ m/is on/;
+            ipmitool('chassis power on');
+        }
+    } else {
+        die('Backend ' . get_var('BACKEND') . 'not supported');
+    }
+}
+
+=head2 poweroff
+
+ poweroff()
+
+Power off the machine if the backend in use is supported. 
+=cut
+sub poweroff {
+    if (check_var('BACKEND', 'ipmi')) {
+        ipmitool("chassis power off");
+        while (1) {
+            sleep(3);
+            my $stdout = ipmitool('chassis power status');
+            last if $stdout =~ m/is off/;
+            ipmitool('chassis power off');
+        }
+    } else {
+        die('Backend ' . get_var('BACKEND') . 'not supported');
+    }
+}
+
+=head2 poweron
+
+ set_net_boot()
+
+Set boot from network if the backend in use is supported. 
+=cut
+sub set_net_boot {
+    if (check_var('BACKEND', 'ipmi')) {
+        while (1) {
+            my $stdout = ipmitool('chassis bootparam get 5');
+            last if $stdout =~ m/Force PXE/;
+            ipmitool("chassis bootdev pxe");
+            sleep(3);
+        }
+    } else {
+        die('Backend ' . get_var('BACKEND') . 'not supported');
+    }
+}
+
+=head2 wait_boot
+
+wait_boot()
+
+Wait for the machine to boot up.
+=cut
+sub wait_boot {
+    my $timeout = shift;
+    if (!defined $timeout) {
+        $timeout = 1800;
+    }
+    if (check_var('BACKEND', 'ipmi')) {
+        select_consolei('sol', await_console => 0);
+        assert_screen('linux-login', $timeout);
+    } else {
+        die('Backend ' . get_var('BACKEND') . 'not supported');
+    }
+}
+
+=head2 host_lock
+
+ host_lock() 
+
+Lock the SUT from the baremetal support service. 
+=cut
+sub host_lock {
+    my $host        = get_required_var('SUT_IP');
+    my $ip          = inet_ntoa(inet_aton($host));
```

Comment:
Not sure what's in `SUT_IP` but maybe we need to check for `undef` on inet_aton() if hostname isn't resolvable